src/app/webui/components/hooks/useChat.ts

```
@@ -0,0 +1,210 @@
+    const lastImageUriRef = useRef<string | null>(null);
+    const [status, setStatus] = useState<'connecting' | 'open' | 'closed'>('connecting');
+
+    useEffect(() => {
+        const ws = new WebSocket(wsUrl);
+        const generateUniqueId = () =>
+            `msg-${Date.now()}-${Math.random().toString(36).substring(2, 9)}`;
+
+        wsRef.current = ws;
+        ws.onopen = () => setStatus('open');
+        ws.onclose = () => setStatus('closed');
+        ws.onmessage = (event: MessageEvent) => {
```

ESLint

[error] 'MessageEvent' is not defined.

(no-undef)